src/wavm/openmp.cpp

```diff
@@ -118,14 +80,67 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
 {
     const std::shared_ptr<spdlog::logger>& logger = faabric::util::getLogger();
     logger->debug("S - omp_set_max_active_levels {}", level);
+
     if (level < 0) {
         logger->warn("Trying to set active level with a negative number {}",
                      level);
-        return;
+    } else {
+        getOpenMPContext().level->maxActiveLevels = level;
     }
-    thisLevel->maxActiveLevel = level;
 }
 
+WAVM_DEFINE_INTRINSIC_FUNCTION(env,
```

Comment:
These functions are unchanged, I've just grouped those related to number of threads together nearer the top.